autoload/lsp/internal/work_done_progress.vim

```diff
@@ -0,0 +1,68 @@
+" https://microsoft.github.io/language-server-protocol/specifications/specification-current/#progress
+
+let s:progress_ui = []
+let s:enabled = 0
+
+function! lsp#internal#work_done_progress#_enable() abort
+    if !g:lsp_work_done_progress_enabled | return | endif
+
+    if s:enabled | return | endif
+    let s:enabled = 1
+
+    let s:Dispose = lsp#callbag#pipe(
```

Comment:
we should reset the state of `s:progress_ui` when we enable and disable.

We need to do this when we enable so we get clear state for the first time. and disable so that when `lsp#get_progress()` is called it gets empty array.

Reply:
this is also useful for tests so that you are guaranteed your state is clean before your tests start.

Reply:
That is true.
I fixed it.